src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -526,6 +528,8 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
                     return false;
                 }
 
+                Debug.Assert(specialElementType.SizeInBytes() is 1 or 2 or 4 or 8);
```

Comment:
Are there no plans for `decimal`, which has a well-defined size and layout and which has constant support in other places?

Or is this possibly a "future consideration"?